game-core/src/main/java/games/strategy/engine/data/MutableProperty.java

```diff
@@ -0,0 +1,147 @@
+package games.strategy.engine.data;
+
+import java.util.function.Supplier;
+
+import javax.annotation.Nullable;
+
+/**
+ * A wrapper interface to Bundle setters, getters and resetters of the same field.
+ *
+ * @param <T> The type of the field to set, get and reset.
+ */
+public class MutableProperty<T> {
+
+  private final ThrowingConsumer<T, Exception> setter;
+  private final ThrowingConsumer<String, Exception> stringSetter;
+  private final Supplier<T> getter;
+  private final Runnable resetter;
+
+  private MutableProperty(
+      @Nullable final ThrowingConsumer<T, Exception> setter,
+      @Nullable final ThrowingConsumer<String, Exception> stringSetter,
+      @Nullable final Supplier<T> getter,
+      @Nullable final Runnable resetter) {
+    this.setter = setter != null ? setter : o -> {
+      throw new UnsupportedOperationException("No Setter has been defined!");
+    };
```

Comment:
This is a step in the right direction, but it would be nice to get rid of the `null` usage.  I submitted a PR against your fork with a suggestion to accomplish this.